### PR TITLE
[Command] add command to generate jwt

### PR DESCRIPTION
**Pull request by SirDomin**
| Q               | A
| --------------- | -----
| Branch?         | master
| Bug fix?        | no
| New feature?    | yes
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | 
| License         | MIT

image

### CODE REVIEW

**Review by AdamKasp — APPROVED** [review]

File: src/Sylius/Bundle/CoreBundle/Command/InstallCommand.php [hunk 1]
```diff
@@ -36,6 +36,10 @@ final class InstallCommand extends AbstractInstallCommand
             'command' => 'setup',
             'message' => 'Shop configuration.',
         ],
+        [
+            'command' => 'jwt_setup',
```
Comment (Zales0123) [inline src/Sylius/Bundle/CoreBundle/Command/InstallCommand.php:40]: `jwt-setup` to be consistent with `check-requirements`

File: src/Sylius/Bundle/CoreBundle/Command/InstallCommand.php [hunk 2]
```diff
@@ -36,6 +36,10 @@ final class InstallCommand extends AbstractInstallCommand
...
+            'message' => 'JWT configuration.',
```
Comment (Zales0123) [inline src/Sylius/Bundle/CoreBundle/Command/InstallCommand.php:41]: As we have "**Checking** system requirements", "**Setting** up the database" and "**Installing** assets" I vote to change it to `Configuring JWT token.`... but then we should also change `Shop configuration.` to `Configuring Shop.` 😄 💃  Nevertheless, consistency is the clear to world peace (or at least code peace)

File: src/Sylius/Bundle/CoreBundle/Command/JwtConfigurationCommand.php [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+namespace Sylius\Bundle\CoreBundle\Command;
+
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Question\ConfirmationQuestion;
+
+final class JwtConfigurationCommand extends AbstractInstallCommand
+{
+    protected static $defaultName = 'sylius:install:jwt_setup';
```
Comment (Zales0123) [inline src/Sylius/Bundle/CoreBundle/Command/JwtConfigurationCommand.php:22]: `sylius:install:jwt-setup`, again, consitency

File: src/Sylius/Bundle/CoreBundle/Command/JwtConfigurationCommand.php [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+EOT
+            )
+        ;
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output): int
+    {
+        $helper = $this->getHelper('question');
+
+        $output->writeln('Configuring API');
```
Comment (Zales0123) [inline src/Sylius/Bundle/CoreBundle/Command/JwtConfigurationCommand.php:40]: `'Generating JWT token for Sylius API'`?

File: src/Sylius/Bundle/CoreBundle/Command/JwtConfigurationCommand.php [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+
+        $question = new ConfirmationQuestion('Do you want to generate JWT token? (y/N)', false);
+
+        if (!$helper->ask($input, $output, $question)) {
+            return 0;
+        }
+
+        $this->commandExecutor->runCommand('lexik:jwt:generate-keypair', ['--overwrite' => true], $output);
+
+        $output->writeln('Please, remember to enable Sylius API');
```
Comment (Zales0123) [inline src/Sylius/Bundle/CoreBundle/Command/JwtConfigurationCommand.php:50]: Maybe we can link there a proper documentation chapter https://docs.sylius.com/en/1.10/book/api/introduction.html?

Comment (Zales0123) [reply src/Sylius/Bundle/CoreBundle/Command/JwtConfigurationCommand.php:50]: Btw, we should probably mention this command somewhere in the documentation as well 😄

**Review by Zales0123 — COMMENTED** [review]
Good idea 👌 But why not put it on `1.10`? I would see it as an improvement to our installation process, not an entirely new feature 🖖 🚀

File: src/Sylius/Bundle/CoreBundle/Command/JwtConfigurationCommand.php (hunk 5 above)
Comment (SirDomin) [reply src/Sylius/Bundle/CoreBundle/Command/JwtConfigurationCommand.php:50]: yeah, added link and mentioned it in installation docs

**Review by GSadee — APPROVED** [review]

**Comment by GSadee** [comment]
Thanks, @SirDomin! :tada:

**Pull request merged**